src/main/java/net/mcreator/generator/LocalizationUtils.java

```diff
@@ -142,7 +154,12 @@ private static void addLocalizationEntry(Generator generator, String key, Map<?,
 			if (prefix != null)
 				value = prefix + value;
 
-			generator.getWorkspace().setLocalization(key, value);
+			if (shouldBeSkippedBasedOnCondition(generator, template, entry)) {
+				// If localization key is skipped, we make sure to remove the localization entry
+				generator.getWorkspace().removeLocalizationEntryByKey(key);

```

Comment:
We can't know if localization entry is used somewhere else, so this should not be used. Same as how we don't delete GUI generated textures or textures of MEs

Reply:
I should just remove the line then? If we don't delete entries, then why are they deleted when we delete mod elements then? (It's a genuine question, just trying to understand, not to argue)

Reply:
That was implemented when there wasn't a procedure block to get text by localization key and did not get reworked with time I think.

Reply:
Instead of blindly removing localization entries, isn't there a way to check if it's already being used before removing them then?

Reply:
Oh nice, I'll wait for it to be merged

Reply:
It can be still used by custom code, but that is on the user I guess.

A valid point @Max094Reikeb with deleting keys when deleting the mod element, all good then with this, you can leave it as it is.